spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -171,10 +173,12 @@
       InputFile location,
       FileScanTask task,
       Schema readSchema) {
+    NameMapping nameMapping = MappingUtil.create(readSchema);
     return Parquet.read(location)
         .project(readSchema)
+        .withNameMapping(nameMapping)
         .split(task.start(), task.length())
-        .createReaderFunc(fileSchema -> SparkParquetReaders.buildReader(readSchema, fileSchema))
+        .createReaderFunc(fileSchema -> SparkParquetReaders.buildReader(readSchema, fileSchema, nameMapping))
```

Comment:
The `fileSchema` passed here should by the type with IDs filled in.

Reply:
Agreed, I think we don't need the third parameter `namemapping` as well.